### ISSUE TITLE
Fake in test bleeds over to other tests

### ISSUE THREAD

**Issue by chadhobson** [adonisjs/adonis-framework#1059]
## Package version
5.09

## Node.js and npm version
Node: 11.13.0
NPM: 6.7.0

## Sample Code (to reproduce the issue)
See foo-test.spec.js in repo

## BONUS (a sample repo to reproduce the issue)
https://github.com/chadhobson/AdonisFakeFailsTest

**Comment by BatuhanK** [comment]
we are facing same issue. @thetutlage if you have time, can you help us to find a solution ?

**Comment by RomainLanz** [comment]
Hey @chadhobson, @BatuhanK! 👋 

This is a well known issue with 4.0 and this is already fix in 5.0.
@thetutlage will see if it's worth it to backport the fix to 4.0 because it requires a lot of changes.

**Comment by thetutlage** [comment]
Yes, a big known bug and should be good with the newer release of IoC container, which relies on proxies vs faking after they have been referenced else where.
